sigmf/sigmffile.py

```diff
@@ -630,9 +630,11 @@ def get_sigmf_filenames(filename):
     Returned as dict with 'data_fn', 'meta_fn', and 'archive_fn' as keys.
 
     Keyword arguments:
-    filename -- the SigMF filename
+    filename -- the SigMF dataset filename (with or without .sigmf* extension)
     """
-    filename = path.splitext(filename)[0]
+    filename_lower = filename.lower()
+    if filename_lower.endswith(SIGMF_DATASET_EXT) or filename_lower.endswith(SIGMF_METADATA_EXT) or filename_lower.endswith(SIGMF_ARCHIVE_EXT):
```

Comment:
`.sigmf-collection` is a valid name here also now - probably need to incorporate that into this somehow.

Reply:
That's true, but I'm not sure how it fits, here, because Collections point to `SigMFFile`s. It may the be case that this code needs to assume a unitary Recording, regardless of whether or not it's part of a collection?

Reply:
I suppose the thinking here is that we will need a way to reference the collection from the file, however in reading your comment there are two issues with my original thought, first collection names are necessarily different from sigmf files, and more importantly the collection file is already an explicit metadata key.

So marking resolved.